package/MDAnalysis/converters/OpenMMParser.py

```diff
@@ -123,24 +147,62 @@ def _mda_topology_from_omm_topology(self, omm_topology):
         bond_orders = [b.order for b in omm_topology.bonds()]
         bond_types = [b.type for b in omm_topology.bonds()]
 
-        n_atoms = len(atomids)
-        n_residues = len(resids)
-        n_segments = len(segids)
-
         attrs = [
             Atomids(np.array(atomids, dtype=np.int32)),
             Atomnames(np.array(atomnames, dtype=object)),
-            Atomtypes(np.array(atomtypes, dtype=object)),
             Bonds(bonds, types=bond_types, order=bond_orders, guessed=False),
             ChainIDs(np.array(chainids, dtype=object)),
-            Elements(np.array(elements, dtype=object)),
-            Masses(np.array(masses, dtype=np.float32)),
             Resids(resids),
             Resnums(resnums),
             Resnames(resnames),
             Segids(segids),
         ]
 
+        validated_elements = []
+        masses = []
+        atomtypes = []
+        warn = False
+        for a in omm_topology.atoms():
+            elem = a.element
+            if elem is not None:
+                if elem.symbol.capitalize() in SYMB2Z:
+                    validated_elements.append(elem.symbol)
+                else:
+                    validated_elements.append('')
+                atomtypes.append(elem.symbol)
+                masses.append(elem.mass.value_in_unit(daltons))
+            else:
+                warn = True
+                validated_elements.append('')
+                masses.append(0.0)
+                atomtypes.append('X')
+
+        if warn:
+            wmsg = (f"Element information missing for some atoms. "
+                    f"These have been given an empty element record "
+                    f"with their atomtype set to 'X' "
+                    f"and their mass set to 0.0. "
+                    f"If needed they can be guessed using "
+                    f"MDAnalysis.topology.guessers.")
```

Comment:
If there's no variable substitution then f-strings aren't worth it.

```suggestion
            wmsg = ("Element information missing for some atoms. "
                    "These have been given an empty element record "
                    "with their atomtype set to 'X' "
                    "and their mass set to 0.0. "
                    "If needed they can be guessed using "
                    "MDAnalysis.topology.guessers.")
```